newsroom/topics/views.py

```diff
@@ -1,83 +1,105 @@
 from bson import ObjectId
+from typing import Optional
+from pydantic import BaseModel, ValidationError
 
 from superdesk.core import json, get_app_config
-from superdesk import get_resource_service
-from superdesk.flask import jsonify, abort, url_for, session
+from superdesk.flask import abort, url_for, session
+from superdesk.core.web import Request, Response
 
 from newsroom.types import Topic
-from newsroom.topics import blueprint
-from newsroom.topics.topics import get_user_topics as _get_user_topics, auto_enable_user_emails
-from newsroom.auth import get_user, get_user_id
+from newsroom.email import send_user_email
 from newsroom.decorator import login_required
+from newsroom.auth import get_user, get_user_id
+from newsroom.topics.topics_async import get_user_topics as _get_user_topics, auto_enable_user_emails
 from newsroom.utils import get_json_or_400, get_entity_or_404
-from newsroom.email import send_user_email
 from newsroom.notifications import (
     push_user_notification,
     push_company_notification,
     save_user_notifications,
     UserNotification,
 )
+from .topics_async import topic_endpoints, TopicService, TopicResourceModel
+from newsroom.users.service import UsersService
+
+
+class RouteArguments(BaseModel):
+    user_id: Optional[str] = None
+    topic_id: Optional[str] = None
 
 
-@blueprint.route("/users/<_id>/topics", methods=["GET"])
+@topic_endpoints.endpoint("/users/<string:user_id>/topics", methods=["GET"])
 @login_required
-async def get_topics(_id):
+async def get_topics(args: RouteArguments, params: None, request: Request):
     """Returns list of followed topics of given user"""
-    if session["user"] != str(_id):
+    if session["user"] != str(args.user_id):
         abort(403)
-    return jsonify({"_items": _get_user_topics(_id)}), 200
+    topics = await _get_user_topics(args.user_id)
+    return Response({"_items": topics})
 
 
-@blueprint.route("/users/<_id>/topics", methods=["POST"])
+@topic_endpoints.endpoint("/users/<string:user_id>/topics", methods=["POST"])
 @login_required
-async def post_topic(_id):
+async def post_topic(args: RouteArguments, params: None, request: Request):
     """Creates a user topic"""
     user = get_user()
-    if str(user["_id"]) != str(_id):
+
+    if not user or str(user["_id"]) != str(args.user_id):
         abort(403)
 
     topic = await get_json_or_400()
-    topic["user"] = user["_id"]
-    topic["company"] = user.get("company")
+
+    if user:
+        data = {
+            "user": user.get("_id"),
+            "company": user.get("company"),
+            "_id": ObjectId(),
+            "created_filter": topic.pop("created", {}),
+            "is_global": topic.get("is_global", False),
+        }
+        topic.update(data)
 
     for subscriber in topic.get("subscribers") or []:
         subscriber["user_id"] = ObjectId(subscriber["user_id"])
 
-    ids = get_resource_service("topics").post([topic])
+    data = TopicResourceModel.model_validate(topic)
```

Comment:
No need to convert the dict to a ResourceModel instance now, you can pass dicts to `create` methods now